modules/virt-creating-a-primary-cluster-udn.adoc

```diff
@@ -0,0 +1,49 @@
+// Module included in the following assemblies:
+//
+// * virt/vm_networking/virt-connecting-vm-to-primary-udn.adoc
+
+:_mod-docs-content-type: PROCEDURE                                
+[id="virt-creating-a-primary-cluster-udn_{context}"]                                  
+= Creating a primary cluster-scoped user-defined network
+
+You can connect multiple namespaces to the same primary user-defined network (UDN) to achieve native tenant isolation.
+
+.Prerequisites
+. You have access to the cluster as a user with `cluster-admin` privileges.
+. You have installed the OpenShift CLI (`oc`).
+
+.Procedure
+. Create a `ClusterUserDefinedNetwork` object to specify the custom network configuration:
++
+.Example `ClusterUserDefinedNetwork` manifest
+[source,yaml]
+----
+kind: ClusterUserDefinedNetwork
+metadata:
+  name: cudn-l2-net # <1>
+spec:
+  namespaceSelector: # <2>
+    matchExpressions:
+    - key: kubernetes.io/metadata.name
+      operator: In
+      values:
+        - "mynamespace"
+        - "theirnamespace"
+  template:
+    topology: Layer2 # <3>
+    role: Primary # <4>
+    mtu: 9000
+    subnets: ["10.0.0.0/24"]
+    excludeSubnets: ["10.0.0.100/26"]
+----
+<1> The name of the `ClusterUserDefinedNetwork` custom resource.
+<2> Specifies which namespaces can connect to this network. You can use the `kubernetes.io/metadata.name` label to ensure that only namespaces with the matching label are able to access the network.
+<3> The topological configuration of the network. The required value is `Layer2`. A `Layer2` topology creates a logical switch that is shared by all nodes.
+<4> Specifies if the UDN is primary or secondary. {VirtProductName} only supports the `Primary` role. This means that the UDN acts as the primary network for the VM and all default traffic passes through this network.
+
+. Apply the `ClusterUserDefinedNetwork` manifest by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f <filename>.yaml
```

Comment:
Please consider spesifying `--validate` flag in `oc` commands around UDN & CUDN CRDs to ensure strict validation is done.
It will encourage users to use this flag avoiding invalid yaml create silently.

Reply:
Neat suggestion.